magma/lib/magma/server/query.rb

```diff
@@ -14,7 +14,10 @@ def action
         show_disconnected: @params[:show_disconnected],
         restrict: !@user.can_see_restricted?(@project_name),
         user: @user,
-        timeout: Magma.instance.config(:query_timeout)
+        timeout: Magma.instance.config(:query_timeout),
+        page: @params[:page],
```

Comment:
This should work as expected, at least for the top-level model being queried. `Magma::Question` already takes these arguments and uses them for the `/retrieve` endpoint, so this just passes them through via the `/query` endpoint.

Reply:
Does it need page_size?